src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1525,1898 @@ interface MPSMatrixMultiplication {
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(NSObject), Name = "MPSNNDefaultPadding")]
+	interface MPSNnDefaultPadding : MPSNnPadding
+	{
+		[Static]
+		[Export ("paddingWithMethod:")]
+		MPSNnDefaultPadding PaddingWithMethod (MPSNnPaddingMethod method);
+
+		[Static]
+		[Export ("paddingForTensorflowAveragePooling")]
+		MPSNnDefaultPadding PaddingForTensorflowAveragePooling ();
+	}
+	
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSKernel), Name = "MPSCNNBinaryKernel")]
+	interface MPSCnnBinaryKernel
+	{
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+
+		[Export ("primaryOffset", ArgumentSemantic.Assign)]
+		MPSOffset PrimaryOffset { get; set; }
+
+		[Export ("secondaryOffset", ArgumentSemantic.Assign)]
+		MPSOffset SecondaryOffset { get; set; }
+
+		[Export ("clipRect", ArgumentSemantic.Assign)]
+		MTLRegion ClipRect { get; set; }
+
+		[Export ("destinationFeatureChannelOffset")]
+		nuint DestinationFeatureChannelOffset { get; set; }
+
+		[Export ("primaryEdgeMode", ArgumentSemantic.Assign)]
+		MPSImageEdgeMode PrimaryEdgeMode { get; set; }
+
+		[Export ("secondaryEdgeMode", ArgumentSemantic.Assign)]
+		MPSImageEdgeMode SecondaryEdgeMode { get; set; }
+
+		[Export ("kernelWidth")]
+		nuint KernelWidth { get; }
+
+		[Export ("kernelHeight")]
+		nuint KernelHeight { get; }
+
+		[Export ("primaryStrideInPixelsX")]
+		nuint PrimaryStrideInPixelsX { get; }
+
+		[Export ("primaryStrideInPixelsY")]
+		nuint PrimaryStrideInPixelsY { get; }
+
+		[Export ("secondaryStrideInPixelsX")]
+		nuint SecondaryStrideInPixelsX { get; }
+
+		[Export ("secondaryStrideInPixelsY")]
+		nuint SecondaryStrideInPixelsY { get; }
+
+		[Export ("isBackwards")]
+		bool IsBackwards { get; }
+
+		[Export ("padding", ArgumentSemantic.Retain)]
+		IMPSNnPadding Padding { get; set; }
+
+		[Export ("destinationImageAllocator", ArgumentSemantic.Retain)]
+		IMPSImageAllocator DestinationImageAllocator { get; set; }
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[Export ("encodeToCommandBuffer:primaryImage:secondaryImage:destinationImage:")]
+		void Encode (IMTLCommandBuffer commandBuffer, MPSImage primaryImage, MPSImage secondaryImage, MPSImage destinationImage);
+
+		[Export ("encodeToCommandBuffer:primaryImage:secondaryImage:")]
+		MPSImage EncodeToCommandBuffer (IMTLCommandBuffer commandBuffer, MPSImage primaryImage, MPSImage secondaryImage);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnNeuron), Name = "MPSCNNNeuronPReLU")]
+	interface MPSCnnNeuronPReLU
+	{
+		[Export ("initWithDevice:a:count:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, IntPtr /* float* */ a, nuint count);
```

Comment:
I think the code-behind implementation is sufficient, so `[Internal]` to hide the IntPtr signature.